File: entsoe-client/src/main/java/org/example/model/common/Direction.java
package org.example.model.common;

import java.util.Arrays;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public enum Direction {
  UP("A01", "Up"),
  DOWN("A02", "Down"),
  SYMMETRIC("A03", "Symmetric");

  private final String code;
  private final String description;

  public static Direction fromCode(String code) {
    return Arrays.stream(values())
        .filter(type -> type.code.equals(code))
        .findFirst()
        .orElseThrow(() -> new IllegalArgumentException("Unknown Direction code: " + code));
  }
}

File: entsoe_client/src/entsoe_client/model/common/direction.py
from enum import Enum
from typing import Self

from src.entsoe_client.exceptions.unknown_direction_error import (
    UnknownDirectionError,
)


class Direction(Enum):
    UP = ("A01", "Up")
    DOWN = ("A02", "Down")
    SYMMETRIC = ("A03", "Symmetric")

    def __init__(self, code: str, description: str) -> None:
        self.code = code
        self.description = description

    @classmethod
    def from_code(cls, code: str) -> Self:
        for member in cls:
            if member.code == code:
                return member
        raise UnknownDirectionError(code)
